src/gallium/drivers/lima/ir/pp/node_to_instr.c

```diff
@@ -37,6 +37,53 @@ static bool create_new_instr(ppir_block *block, ppir_node *node)
    return true;
 }
 
+static bool insert_to_load_tex(ppir_block *block, ppir_node *load_coords, ppir_node *ldtex)
+{
+   ppir_dest *dest = ppir_node_get_dest(ldtex);
+   ppir_node *move = NULL;
+
+   ppir_load_node *load = ppir_node_to_load(load_coords);
+   load->dest.type = ppir_target_pipeline;
+   load->dest.pipeline = ppir_pipeline_reg_discard;
+
+   ppir_load_texture_node *load_texture = ppir_node_to_load_texture(ldtex);
+   load_texture->src_coords.type = ppir_target_pipeline;
+   load_texture->src_coords.pipeline = ppir_pipeline_reg_varying;
```

Comment:
Why adding a new pipeline reg here? I think you can just use ppir_pipeline_reg_discard as it's the dest of load_coord.

Reply:
Because 'ppir_pipeline_reg_discard' is misleading as src reg. We're getting coords from varying.

Reply:
My original thought is load_texture->src_coords gets input from load_coord->dest, so only set load_coord->dest make them miss-match. If we make different pipeline reg, they miss match again. If you think the name of ppir_pipeline_reg_discard cause misleading, we can change it to ppir_pipeline_reg_texcoord or other better name you can think of.

Reply:
OK, I don't see a solution that I like here, so I'll just make it as you proposed initially :)

I pushed fixed version to my branch.